Realm/RLMRealm.mm

```diff
@@ -706,6 +706,53 @@ - (void)cancelWriteTransaction {
     }
 }
 
+#ifdef REALM_ASYNC_WRITES
+
+- (BOOL)inAsyncWriteTransaction {
+    return _realm->is_in_async_transaction();
+}
+
+- (AsyncHandle)beginAsyncWriteTransaction:(void(^)())block {
+    return _realm->async_begin_transaction(block);
+}
+
+- (AsyncHandle)commitAsyncWriteTransaction {
+    return _realm->async_commit_transaction();
+}
+
+- (AsyncHandle)commitAsyncWriteTransaction:(nullable void(^)())doneBlock {
+    if (doneBlock) {
+        return _realm->async_commit_transaction(doneBlock);
+    }
+    return _realm->async_commit_transaction();
+}
+
+- (AsyncHandle)commitAsyncWriteTransaction:(nullable void(^)())doneBlock isGroupingAllowed:(BOOL)isGroupingAllowed {
+    if (doneBlock) {
+        return _realm->async_commit_transaction(doneBlock, isGroupingAllowed);
+    }
+    return _realm->async_commit_transaction(nil, isGroupingAllowed);
+}
+
+- (void)cancelAsyncTransaction:(AsyncHandle)handle {
+    _realm->async_cancel_transaction(handle);
+}
+
+- (void)asyncTransactionWithBlock:(void(^)())block onComplete:(nullable void (^)())doneBlock {
+    AsyncHandle transaction = [self beginAsyncWriteTransaction: ^{
+        block();
+        if (_realm->is_in_async_transaction()) {
```

Comment:
This looks like there is undefined behavior?  How could they not be in an async transaction and if they aren't, should it throw instead of never committing?

Reply:
You're right. It's something I added looking at sync transactions